### PR TITLE
implement search bar in shortcuts menu

### LINKED ISSUE #3648: Include search/filter to the custom shortcuts settings table
After #3641, custom shortcuts changed to be a nice table.

It would be very handy if the table would be searchable just like in the manage sets dialog or cards in the deck editor...

### PR DESCRIPTION
**Pull request by RickyRister**
## Related Ticket(s)
- Fixes #3648

## What will change with this Pull Request?

https://github.com/user-attachments/assets/e39e3451-389c-4e2e-b68d-9e10b40fb1ad

Shortcut settings menu now has a search bar.

Currently the search bar is a very primitive implementation. It searches for that exact string, including spaces. It doesn't do the thing where it splits on spaces and shows the row if it matches any word in the string.
Perhaps we can improve the functionality of the search bar later.

- Moved the Shortcuts table from a `QTreeWidget` into its own `QTreeView` subclass in a separate file
  - Also moved the view and model logic into the subclass.

## Screenshots
Screenshot 2024-12-21 at 2 53 12 PM

Screenshot 2024-12-21 at 3 03 02 PM

### CODE REVIEW

**Review by ZeldaZach — APPROVED** [review]

File: cockatrice/src/dialogs/dlg_settings.cpp [hunk 1]
```diff
@@ -1251,13 +1251,24 @@ void SoundSettingsPage::retranslateUi()
 
 ShortcutSettingsPage::ShortcutSettingsPage()
 {
+    // search bar
+    searchEdit = new SearchLineEdit;
+    searchEdit->setObjectName("searchEdit");
+    searchEdit->setPlaceholderText(tr("Search by shortcut name"));
```
Comment (ebbit1q) [inline cockatrice/src/dialogs/dlg_settings.cpp:1257]: translateable text should go in retranslateUI

File: cockatrice/src/settings/shortcut_treeview.h [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+private:
+    QStandardItemModel *shortcutsModel;
+    QSortFilterProxyModel *proxyModel;
+    void populateShortcutsModel();
+
+private slots:
+    void refreshShortcuts();
+
+protected:
```
Comment (ebbit1q) [inline cockatrice/src/settings/shortcut_treeview.h:30]: what does setting this as protected actually do for this class that isn't subclassed again and isn't part of a library we provide (we don't provide any)?

Comment (RickyRister) [reply cockatrice/src/settings/shortcut_treeview.h:30]: We’re overriding a protected method in the parent class

File: cockatrice/src/dialogs/dlg_settings.cpp (hunk 1 above)
Comment (RickyRister) [reply cockatrice/src/dialogs/dlg_settings.cpp:1257]: Oops… feel free to open a PR to fix it

Comment (ZeldaZach) [reply cockatrice/src/dialogs/dlg_settings.cpp:1257]: Fixing

File: cockatrice/src/settings/shortcut_treeview.h (hunk 2 above)
Comment (ebbit1q) [reply cockatrice/src/settings/shortcut_treeview.h:30]: ah, I see

**Pull request merged**